resources/analysisTools/copyNumberEstimationWorkflow/beagle_embed_haplotypes_vcf.py

```diff
@@ -0,0 +1,54 @@
+ #!/usr/bin/python
+import argparse
+import gzip
+parser = argparse.ArgumentParser()
+parser.add_argument('--hap_file', help='Input file with phased haplotypes (i.e. output from Beagle)')
+parser.add_argument('--vcf_file', help='Input .vcf file')
+parser.add_argument('--out_file', help='Out .vcf file')
+args = parser.parse_args()
+
+if args.hap_file.split('.')[-1] == 'gz':
+    hap_file = gzip.open( args.hap_file, "r" )
+else:
+    hap_file = open( args.hap_file, "r" )
+vcf_file = open( args.vcf_file, "r" )
+outfile = open( args.out_file, "w" )
+
+hap_line = hap_file.readline()
+while hap_line[0] == "#":
+    hap_line = hap_file.readline()
+
+if hap_line:
+    hap_line = hap_line.rstrip().split()
+
+for vcf_line in vcf_file:
+    
+    if vcf_line[0] != "#":
+        
+        vcf_line = vcf_line.rstrip().split("\t")
+        vcf_line[0] = vcf_line[0].replace('chr', '')
+        vcf_line[0] = vcf_line[0].replace( 'X', '23' )
+
+        if len(vcf_line) >= 9 and hap_line:
```

Comment:
This does not actually check, that the line is really the 'sample0' line that you added? But that is expected, or not?

If so, please at least check that the header[9]=='sample0' and fail if not. 

Like before consider using a column index variable (SAMPL0_COL=9), or a (colname => colindex) map. The latter solution would allow you to index also other columns like

```python
hap_line[colMap['sample0']]
```

which is of course more verbose, but *much* more understandible. Also Python dies with a descriptive error message if you try to access a missing key, which gives you a free check for whether the input is as expected and thus greatly improved code quality/robustness.

Reply:
As the other helper functions duplicates the sample it should be expected that this works. I added an assertion to check that the header of the last column is called `sample0`.
One could furthermore test that `vcf_line[9]` and `vcf_line[10]` are the same but might be an overkill.

I also replaced `[8]` by `[GT_COL]` and `[9]` by `[SAMPLE_COL]` for enhanced readability.